generic3g/MAPL_Generic.F90

```diff
@@ -489,6 +499,18 @@ subroutine gridcomp_add_fieldspec( &
       component_spec => outer_meta%get_component_spec()
       call component_spec%var_specs%push_back(var_spec)
 
+      if (present(add2export)) then
```

Comment:
This is fine, but on a meta level, I see that this procedure is getting awfully large for a layer that was intended to be thin wrappers.   We may need to refactor soon.